src/azul/queues.py

```diff
@@ -430,19 +433,24 @@ def manage_lambdas(self, queues: Mapping[str, Queue], enable: bool):
             def submit(f, *args, **kwargs):
                 futures.append(tpe.submit(f, *args, **kwargs))
 
+            indexer = load_app_module('indexer')
+            lambdas = [
+                indexer.contribute,
+                indexer.contribute_retry,
+                indexer.aggregate,
+                indexer.aggregate_retry
+            ]
+            queues_to_lambda = {lambda_.queue: lambda_ for lambda_ in lambdas}
             for queue_name, queue in queues.items():
                 if queue_name == config.notifications_queue_name():
+                    # Since the index lambda writes to the notifications queue,
+                    # we must deactivate it also.
                     submit(self._manage_lambda, config.indexer_name, enable)
-                    submit(self._manage_sqs_push, config.indexer_name + '-contribute', queue, enable)
-                elif queue_name == config.notifications_queue_name(retry=True):
-                    submit(self._manage_sqs_push, config.indexer_name + '-contribute_retry', queue, enable)
-                elif queue_name == config.tallies_queue_name():
-                    submit(self._manage_sqs_push, config.indexer_name + '-aggregate', queue, enable)
-                elif queue_name == config.tallies_queue_name(retry=True):
-                    # FIXME: Brittle coupling between the string literal below and
-                    #        the handler function name in app.py
-                    #        https://github.com/DataBiosphere/azul/issues/1848
-                    submit(self._manage_sqs_push, config.indexer_name + '-aggregate_retry', queue, enable)
+                if queue_name in queues_to_lambda:
+                    lambda_name = f'{config.indexer_name}-{queues_to_lambda[queue_name].lambda_name}'
+                    submit(self._manage_lambda, lambda_name, enable)
+                else:
+                    assert 'fail' in queue_name
```

Comment:
Ideally we would have a method in `Config` to construct this name without having to specify the separator (`'-'`) here, but no existing method seems to accomplish this. I suspect `Config.qualified_resource_name` will need to be modified analogous to how `unqualified_resource_name` had to be changed to accommodate these same lambda names in https://github.com/DataBiosphere/azul/issues/2731, but I'm not sure how it should be done.

Reply:
I spent some time thinking about the best way to do this, but the fifo queues suffix prevents a neat solution. Since this is the only usage, I'm tempted to leave this as it is. We can see if Hannes thinks it's worth adding.